cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/ExcludedBrokersForLeadershipTest.java

```diff
@@ -112,6 +116,38 @@
     p.add(params(5, RackAwareDistributionGoal.class, excludeB1, OptimizationFailureException.class, DeterministicCluster.rackAwareUnsatisfiable(),
                  deadBroker0, null));
 
+    // ============MinTopicLeadersPerBrokerGoal============
+    Properties configOverrides = new Properties();
+    configOverrides.put(AnalyzerConfig.TOPICS_WITH_MIN_LEADERS_PER_BROKER_CONFIG, TestConstants.TOPIC_MUST_HAVE_LEADER_REPLICAS_ON_BROKERS);
+
+    // With single excluded broker, no dead brokers (No exception, Expected to look optimized)
```

Comment:
Are there any tests showing that there are no dead brokers, but there is an optimization failure exception due to inability to distribute leadership over eligible brokers?

Maybe for that test, we can have a clustermodel with a single relevant topic with 1 leader and multiple brokers?

Reply:
Yes, there is a test case in `ReplicationFactorChangeTest` for this goal to cover what you described:
```
there are no dead brokers, but there is an optimization failure exception due to inability to distribute leadership over eligible brokers. 
```

Reply:
@Lincong But is there a test with excluded brokers with leadership moves, which are tested as part of this class?